### DISCUSSION TITLE
Forecast low alarm and simulation prediction issue in XDrip Follower

### DISCUSSION THREAD

**Discussion by Azucarrr** [Navid200/xDrip#53] [Libre]
Hi, 
I am using Libre2 with the patched Libre App and the nightly release from 24/04/2021. I noticed that my XDrip Sync Follower shows a different (in my opinion wrong) forecast low alarm and a different predictive simulation from master/collector even though all the predictive simulation and alarm settings, like "Alarm at Forecasted Low mins", are the same (copied through QR code). See pictures below. I also noticed that the follower won't display the raw values (white data points) and that the noise differs from collector/master, which can be seen due to different filtered plots (gray lines, can be turned on under XDrip+ Display Settings/Graph Settings/Display filtered plot). 

I read a comment from @jamorham (at gitter.im/jamorham/xdrip+) concerning the LimiTTer back in 2016 saying that the features "low prediction" and "noise calculation" work best if raw and filtered data is available. Since I saw that in the follower no raw data will be displayed and the filtered plot differs from the master (which I think is weird), I think this is an issue/bug.

Is this a known bug? Does anyone know this behavior and possible cause?
 
Master, pic 1:
![Screenshot_20210725-163801](https://user-images.githubusercontent.com/87785433/126903291-c430968d-29dd-4541-aabb-611d654f74ca.png)
 
Master, pic 2: Filtered plot (grey line) is very similar to "normal" plot (yellow, blue data points). No noise information is displayed even though setting is on (Show Noise workings) 
![Screenshot_20210725-163835](https://user-images.githubusercontent.com/87785433/126903310-df16ee7b-8257-4d60-b162-6b59f5c7219a.png)

Master, pic 3: simulation prediction (purple line) reaches aprox. 110mg/dl and forecast low (red line) aprox. 98mg/dl at 5:40pm
![Screenshot_20210725-163839](https://user-images.githubusercontent.com/87785433/126903259-d19eba77-c5b3-4a6e-a846-ec60ac4e8024.png)

XDrip Follower: Filtered plot (grey line) is not similar to "normal" plot (yellow, blue data points), sensor noise is 31.4, simulation prediction (purple line) reaches aprox. 120mg/dl and forecast low (red line) aprox. 125 mg/dl at 5:40pm
![Screenshot_20210725-163847](https://user-images.githubusercontent.com/87785433/126903486-c1b5c5e0-4cdd-436c-a068-3b8b8c500c68.png)

**Comment by Navid200** [comment]
I'm going to focus on the forecast low alert here.
After we talk about it, if you still have questions about Libre-specific matters, I can ask individuals who can help to step in.

Neither master nor follower predicts a low here.
When that happens, xDrip shows text on main screen starting with green.  When the predicted low time moves closer in time, it turns yellow.  When the alert triggers, the text turns red.

In order to compare apples to apples, you need to have the same readings on screen of both master and follower.
When the master says "0 minutes ago", it means the master just got a reading (less than a minute ago).  It may take a while, sometimes close to a minute, for the new readings to be received by the follower.  The forecast low alert relies on extrapolation.  If the set of points is not identical, you don't have a valid comparison.

To address those two concerns, would you please do the following?
First, please enable the forecast low alert.
Then, wait for a time when the alert shows the text on screen in any color.  Then, wait for the follower to receive the last point that the master has received.  Only then, please capture what the two are showing.
If they are different, I would love to see.
